w_emitter.h

```diff
@@ -486,7 +486,15 @@ class WEmitter
 
             for (Decl* field : ut->fields_)
             {
-                // TODO: Do not copy over size.
+                /*
+                 * Fall through if attrs_->is_size_or_count_ t is true,
+                 * which indicates that the struct member is used by the size or
+                 * count attribute, Doing so effectively prevent the callee to
+                 * modify the value of size or count set by the caller.
+                 */
+                if (field->attrs_ && field->attrs_->is_size_or_count_)
+                    continue;
```

Comment:
Instead of falling through, can we check and assert that the the rhs size/count is less than or equal to lhs size/count.

Reply:
Do you think we should check is rhs size/count is not equal to lhs size/count instead? It may be confused that we don't error out the case of rhs size count <= lhs size/count, but the value of lhs size/count will still not be modified by rhs.

Reply:
It is OK not to error out when rhs size/count < lhs size/count. 
In such a case, we just need to copy only rhs size/count number of elements rather than the original lhs size/count.

This is similar to string in/out parameters where the user can reduce the string length, but we don't error out.
sgx_edger8r does not error out as well in such a case even though lhs size/count is not overwritten.

Reply:
Currently, what this PR does is ignoring the rhs size/count set by the callee.

Does your response imply that we need to update this PR so that the lhs size/count can be modified by if the value is less than or equal to the rhs value?

The two design are a bit different and need to be clearly documented. The first design is simpler: the size/count variable is read-only. However, the second one implies that the size/count variable acts as a value-result argument; i.e., the caller should expect to check the returned value.

Reply:
No. My suggestion is to do what Intel edger8r does:
- Keep the count unchanged ie ignore rhs size/count
- The difference is that we currently copy back lhs count number of items;
  intel edger8r only copies back rhs count items.

Reply:
Ideally, we'd need to assign RHS val to a local variable to avoid TOCTOU.
But since we don't use RHS val, it doesn't matter for now.